### PR TITLE
Iss18 v2

### PR DESCRIPTION
**Pull request by ywx649999311**
I think this should work, I tried to test it on descut_test, but the task message got lost occasionally(maybe half, half). It took me half a day to figure out it is because we have two workers connected to redis, if the new task is sent the worker on descut_all, nothing happens on descut_test, but creating error on descut_all (new task getList is not registered there). 

Anyways, everything works great on my Mac, so I think it is ready to be merged.

### CODE REVIEW

**Comment by ywx649999311** [comment]
screen shot 2016-10-02 at 11 12 04 am
This is how the body of data looks like. Basically beyond providing positions, they can filter the list by ccdnum, band and nite.

**Comment by ywx649999311** [comment]
@mgckind note the way the bands option matches the format of the ra and dec, I have also updated the format of band in post job api.

**Comment by mgckind** [comment]
@ywx649999311 great, I'll take a look at the changes. There is one extra thing that its very useful and that is the EXPNUM, which is the exposure number (which is a single band, 70 ccds observation)
adding it would be awesome, so you can filter by expnum and ccd (expnum is used frequently within DB) Thanks!

**Comment by ywx649999311** [comment]
@mgckind ok, I will add it and let you know when it done, then you can review it again. thanks

**Comment by ywx649999311** [comment]
Request body:
screen shot 2016-10-04 at 10 53 37 am
Return:
screen shot 2016-10-04 at 10 53 47 am

@mgckind  Expnum is ready

**Comment by mgckind** [comment]
I saw it, great! I'll take another look

File: api.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ def post(self):
         # validate bands and convert the correct format
         if response['status'] == 'ok':
             if 'band' in arguments:
-                bands = arguments['band'].replace('[','').replace(']','')
+                bands = arguments['band'].replace('[','').replace(']','').replace("'",'').replace(' ','')
```
Comment (ywx649999311) [inline api.py]: yes, it will change to API. It may not work with the old API. I just thought this way seems more intuitive.

Comment (ywx649999311) [inline api.py]: Great, thanks!

File: cutout_cmd/mongo_util.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 from bson.json_util import dumps
```
Comment (mgckind) [inline cutout_cmd/mongo_util.py]: why this change here?

File: static/elements/des_elements/des-login.html [hunk 3]
```diff
@@ -37,12 +37,22 @@
...
+
       properties : {
         error :{
         type: String,
           value: '',
         }
       },
+
+      keyBindings:{
+        'enter:keydown':'_clickSubmit',
```
Comment (mgckind) [inline static/elements/des_elements/des-login.html]: this is great, does it work fine?

File: cutout_cmd/mongo_util.py (hunk 2 above)
Comment (ywx649999311) [inline cutout_cmd/mongo_util.py]: you mean the one below? connect=False? 
with this arg, Mongo throws errors, from the warnings given, I made this change. With this command, the connection will be initiated at the first operation

File: static/elements/des_elements/des-login.html (hunk 3 above)
Comment (ywx649999311) [inline static/elements/des_elements/des-login.html]: yes, I works

Comment (mgckind) [inline static/elements/des_elements/des-login.html]: 👍

File: cutout_cmd/mongo_util.py (hunk 2 above)
Comment (ywx649999311) [inline cutout_cmd/mongo_util.py]: The warning was I should not create a connection before forking

File: api.py [hunk 4]
```diff
@@ -412,7 +412,124 @@ def delete(self):
         self.flush()
         self.finish()
 
+class MongoHandler(tornado.web.RequestHandler):
+
+    @tornado.web.asynchronous
+    def get(self):
```
Comment (mgckind) [inline api.py]: I think this should be a post, especially if we are using a body as part of the request

Comment (ywx649999311) [inline api.py]: just change it to post? or we need another get method at the same time.

File: api.py [hunk 5]
```diff
@@ -412,7 +412,124 @@ def delete(self):
...
+            print (options)
+            try:
+                dtasks.getList.apply_async(args=[df_pos, options, bands, noBlacklist], task_id=qTiid)
+            except Exception as e:
+                response['status']='error'
+                response['message']=str(e)
+            else:
+                res = AsyncResult(qTiid)
+                result = res.wait(5)
+                rt_str = io.StringIO(result)
```
Comment (mgckind) [inline api.py]: what are these 2 lines doing ? its writing the request into a file so celery can read it?

File: cutout_cmd/mongo_util.py (hunk 2 above)
Comment (mgckind) [inline cutout_cmd/mongo_util.py]: I see, thanks

File: api.py (hunk 4 above)
Comment (mgckind) [inline api.py]: I don't think we need a get method, just post, its safer while the data is not public

File: api.py (hunk 5 above)
Comment (ywx649999311) [inline api.py]: it is waiting for the getList task finished and then write the query result into a StringIO object for pandas to read from. Then select the columns we want from the pandas dataframe and export it to another StringIO. The finalIO.getvalue() is the json string containing the data we wanted. Last, I used ast.literal_eval to convert string to a list.

I know it is messy, there are some intermediate step we can remove if we want everything returned from mongo

File: api.py (hunk 4 above)
Comment (ywx649999311) [inline api.py]: ok

File: api.py (hunk 5 above)
Comment (mgckind) [inline api.py]: I got it now, I think is fine maybe we can write directly into a JSON object in the future

Comment (ywx649999311) [inline api.py]: I just made an update, there is no more cvs, but I am still using the StringIO. The get method has been change to post

File: api.py (hunk 1 above)
Comment (mgckind) [inline api.py]: It worked without changes, Its because bands are strings anyway. so 👍

**Comment by mgckind** [comment]
I will try the Mongo API soon, and update the API Help as well. #22   Thanks!
